baekeau/cubed
# Feature/input system – Implement new Input System for cross-platform support

## Summary
This PR introduces a new input system using Unity's Input System package, replacing the legacy Input Manager. This change provides better support for multiple platforms and input devices, including keyboard/mouse, gamepads, and touch screens.

## Problem
Our current input handling relies on Unity's old Input Manager, which lacks flexibility for different input devices and makes it difficult to support multiple platforms efficiently. It also doesn't provide easy remapping of controls for player customization.

## Solution
Implemented a new input system using Unity's Input System package. This solution offers:
- Unified input handling across different devices and platforms
- Easier input remapping and player customization
- Improved performance and reduced input latency
- Better support for new input devices in the future

## Changes
- Added `PlayerInput` reference in the prefab's `PlayerController` 
- Created `GameInput` asset to define input actions and bindings
- Implemented `InputReader` script to manage input events and communicate with game systems
- Updated player movement, camera control, and UI navigation to use the new input system
- Removed old input-related code and dependencies on `Input.GetAxis()` and similar methods

## Testing
1. Open the `TitleScene` in the editor
2. Enter Play mode and verify basic movement and actions with keyboard/mouse
3. Connect a gamepad and ensure all actions work correctly
4. Test that UI scheme is being loaded with pause button

## Notes
- This change will require updates to all scripts that previously used `Input.GetKey()`, `Input.GetButton()`, etc.
- The `InputActions` asset may need further refinement as we implement more game features
- Consider creating an input preset system for different control schemes in the future
- Some commits using ShaderGraph practice was included, but should not affect Input

## Documentation
- Not applicable
